ui/src/main/webapp/src/app/reports/source/source-report.component.ts

```diff
@@ -44,40 +43,49 @@ export class SourceReportComponent implements OnInit, AfterViewChecked {
                 private classificationService: ClassificationService,
                 private hintService: HintService,
                 private notificationService: NotificationService,
-                private http: Http,
                 private _graphJsonToModelService: GraphJSONToModelService<any>
     ) { }
 
     ngOnInit(): void {
-        this.route.params.forEach((params: Params) => {
+        // This multiple route parent searching is kind of ridiculous, but I'm not sure of a better way at the moment?
```

Comment:
We can use `RouteFlattenerService`. You are right, it is ridiculous, I will send PR with `RouteFlattenerService` used